homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,212 @@
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_AGENCY, CONF_ROUTE, CONF_STOP, DOMAIN
+from .util import UserConfig
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _dict_to_select_selector(options: dict[str, str]) -> SelectSelector:
+    return SelectSelector(
+        SelectSelectorConfig(
+            options=[
+                SelectOptionDict(value=key, label=value)
+                for key, value in options.items()
+            ],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    )
+
+
+def _get_agency_tags(client: NextBusClient) -> dict[str, str]:
+    return {a["tag"]: a["title"] for a in client.get_agency_list()["agency"]}
+
+
+def _get_route_tags(client: NextBusClient, agency_tag: str) -> dict[str, str]:
+    return {a["tag"]: a["title"] for a in client.get_route_list(agency_tag)["route"]}
+
+
+def _get_stop_tags(
+    client: NextBusClient, agency_tag: str, route_tag: str
+) -> dict[str, str]:
+    return {
+        a["tag"]: a["title"]
+        for a in client.get_route_config(route_tag, agency_tag)["route"]["stop"]
+    }
+
+
+class NextBusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Nextbus configuration."""
+
+    VERSION = 1
+
+    _agency_tags: dict[str, str]
+    _route_tags: dict[str, str]
+    _stop_tags: dict[str, str]
+
+    def __init__(self):
+        """Initialize NextBus config flow."""
+        self.data: dict[str, str] = {}
+        self._client = NextBusClient(output_format="json")
+        _LOGGER.info("Init new config flow")
+
+    async def async_step_import(self, config_input: UserConfig) -> FlowResult:
+        """Handle import of config."""
+        agency = config_input[CONF_AGENCY]
+        route = config_input[CONF_ROUTE]
+        stop = config_input[CONF_STOP]
+
+        agency_tags = await self.hass.async_add_executor_job(
+            _get_agency_tags, self._client
+        )
+        if agency not in agency_tags:
+            return self.async_abort(reason="invalid_agency")
+
+        route_tags = await self.hass.async_add_executor_job(
+            _get_route_tags, self._client, agency
+        )
+        if route not in route_tags:
+            return self.async_abort(reason="invalid_route")
+
+        stop_tags = await self.hass.async_add_executor_job(
+            _get_stop_tags, self._client, agency, route
+        )
+        if stop not in stop_tags:
+            return self.async_abort(reason="invalid_stop")
+
+        self.data = {
+            CONF_AGENCY: agency,
+            CONF_ROUTE: route,
+            CONF_STOP: stop,
+        }
+
+        # Abort if duplicate entries exist
+        self._async_abort_entries_match(self.data)
+
+        return self.async_create_entry(
+            title=(
+                config_input.get(CONF_NAME)
+                or f"{config_input[CONF_AGENCY]} {config_input[CONF_ROUTE]}"
```

Comment:
I think this should match the title from user flows.
```suggestion
                or f"{agency_tags[agency]} {route_tags[route]} {stop_tags[stop]}"
```

Reply:
It doesn’t match because I thought it better to match existing behavior since this is for migration.

Reply:
I see... Yes since there is no unique id it makes sense.

Anyway that makes me realize that it would be good to migrate to attributes in a preliminary PR:
- self._attr_name = ...
- self._attr_native_value = ...
- self._attr_extra_state_attributes = ...

Reply:
I'm a bit unclear how I should be adding the unique ID here. I got feedback in the other PR that I shouldn't use the Agency + Route + Stop, which should be unique in that if you pass those same values to another sensor, you'd end up with a duplicate sensor providing the exact same results.

If not using those values, it's not clear to me what the advantage of providing a Unique ID that is not tied to the API request being made. It would effectively allow duplicates anyway.

Regardless, is this a blocker for getting this PR shipped? There aren't that many users here (14 according to the stats page) so I don't want to make perfect the enemy of good here and spend too much time on the config flow setup (which is typically done once) when I have another PR improving the API requests which are done many times a day.

Reply:
@epenet is this good now that we're rebased on the merged attr changes?

Reply:
@home-assistant/core Can you please answer the question about the unique id?

Reply:
Thanks @edenhaus. I added a commit to use the Entry ID as the unique ID as suggested, but I still feel it would be better to use the combination of the three config values since those are indeed point to a unique stop much like lat-long point to a unique location.

Reply:
As we now have a unique_id, I suggest equally setting the title in both flows. Otherwise, it can confuse users that the old ones have no stop in the title.